modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notes/GregCustomRxtNotesTestESTestCase.java

```diff
@@ -0,0 +1,214 @@
+    String resourcePath;
+    String noteName;
+    String noteOverviewHash;
+    String replyOverviewHash;
+    String noteAssetId;
+    String replyAssetId;
+    ESTestCommonUtils esTestCommonUtils;
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+
+    @Factory(dataProvider = "userModeProvider")
+    public GregCustomRxtNotesTestESTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init(userMode);
+        String session = getSessionCookie();
+        noteName = "testNote33";
+        assetType = "wsdl";
+        genericRestClient = new GenericRestClient();
+        queryParamMap = new HashMap<String, String>();
+        headerMap = new HashMap<String, String>();
+        resourcePath = FrameworkPathUtil.getSystemResourceLocation()
+                + "artifacts" + File.separator + "GREG" + File.separator + "json" + File.separator;
+        publisherUrl = automationContext.getContextUrls()
+                .getSecureServiceUrl().replace("services", "publisher/apis");
+        esTestCommonUtils = new ESTestCommonUtils(genericRestClient, publisherUrl, headerMap);
+        resourceAdminServiceClient = new ResourceAdminServiceClient(backendURL, session);
+        addCustomRxt();
+        new ServerConfigurationManager(automationContext).restartGracefully();
+        Thread.sleep(120000);
+        SetTestEnvironment();
+    }
+
+    @Test(groups = {"wso2.greg", "wso2.greg.es"}, alwaysRun = true,description = "Add note to custom rxt asset test")
+    public void addNoteToCustomRxtAsset() throws JSONException, IOException {
+        queryParamMap.put("type", "note");
+        //  https://localhost:9443/publisher/apis/assets?type=note
+        String dataBody = String.format(readFile(resourcePath + "publisherAddNoteRestResource.json")
+                , assetType, "testservice1234", noteName);
+        ClientResponse response =
+                genericRestClient.geneticRestRequestPost(publisherUrl + "/assets",
+                        MediaType.APPLICATION_JSON,
+                        MediaType.APPLICATION_JSON, dataBody
+                        , queryParamMap, headerMap, cookieHeaderPublisher);
+        Assert.assertTrue((response.getStatusCode() == 201),
+                "Wrong status code ,Expected 201 OK ,Received " +
+                        response.getStatusCode());
+        JSONObject responseObj = new JSONObject(response.getEntity(String.class));
+        Assert.assertTrue(responseObj.getJSONObject("attributes").get("overview_note").toString().contains(noteName), "Does not create a note");
+        Assert.assertTrue(responseObj.getJSONObject("attributes").get("overview_resourcepath").toString().contains(assetType),"Fault resource path for note");
+        noteOverviewHash = responseObj.getJSONObject("attributes").get("overview_hash").toString();
+        Assert.assertNotNull(noteOverviewHash);
```

Comment:
If noteOverviewHash is null, we get nullpointer exception in line 109. we don't need to get the string value to check whether it is null.